lib/exgboost/parameters.ex

```diff
@@ -0,0 +1,968 @@
+defmodule EXGBoost.Parameters do
+  @global_params [
+    verbosity: [
+      type: {:custom, EXGBoost.Parameters, :validate_verbosity, []},
+      doc: """
+      Verbosity of printing messages. Valid values are:
+      `:silent`, `:warning`, `:info`, `:debug`.
+      """
+    ],
+    use_rmm: [
+      type: :boolean,
+      default: false,
+      doc: """
+      Whether to use RAPIDS Memory Manager for memory allocation.
+      This This option is only applicable when XGBoost is built (compiled)
+      with the RMM plugin enabled. Valid values are `true` and `false`.
+      """
+    ]
+  ]
+
+  @general_params [
+    booster: [
+      type: {:in, [:gbtree, :gblinear, :dart]},
+      default: :gbtree,
+      doc: """
+      Which booster to use. Valid values are `:gbtree`, `:gblinear`, `:dart`
+          * `:gbtree`: tree-based models
+          * `:gblinear`: linear models
+          * `:dart`: tree-based models with dropouts
+      """
+    ],
+    verbosity: [
+      type: {:custom, EXGBoost.Parameters, :validate_verbosity, []},
+      default: :silent,
+      doc: """
+      Verbosity of printing messages. Valid values are:
+      `:silent`, `:warning`, `:info`, `:debug`
+      """
+    ],
+    validate_parameters: [
```

Comment:
I wouldn't make this an option. You should choose one behavior or the other. I prefer that you perform validation by default

Reply:
So the main reason I am currently allowing this is to allow "power users" to use params that perhaps are not perfectly captured using the validators. I guess it's kind of a fail safe in case a validation is broken, it would at least allow people to pass the exact string params they want

Thinking in particular stuff like ndcg@n, map@n: ‘n’ can be assigned as an integer to cut off the top positions in the lists for evaluation.

Reply:
Plus the default is true